include/range/v3/view/iota.hpp

```diff
@@ -524,6 +524,7 @@ namespace ranges
                 detail::take_exactly_view_<iota_view<Val>, true>>
             ints_fn::operator()(Val from, Val to) const
             {
+                RANGES_EXPECT(to >= from);
```

Comment:
I disagree that we should require `to >= from`, at least for unsigned types. I don't particularly care about `[0, -2)`, since we already have `view::iota(0)` to name that infinite ranges, but `[(1u << 30), 2u)` is a perfectly well-defined range of unsigned integers.

This could easily be `RANGES_EXPECT(UnsignedIntegral<Val>() || to >= from);`